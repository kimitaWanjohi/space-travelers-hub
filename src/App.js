import { BrowserRouter, Routes, Route } from "react-router-dom";
import Navbar from "./components/Navbar";
import Missions from "./components/Missions";
import RocketList from './components/RocketList';

function App() {
  return (
    <BrowserRouter>
      <Navbar />
      <div className="container mx-auto">
        <Routes>
          <Route path="/" element={<RocketList />} />
          <Route path="/missions" element={<Missions />} />
          <Route path="/my-profile" element={<h1>My Profile</h1>} />
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
