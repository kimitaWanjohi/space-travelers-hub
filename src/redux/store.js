import { configureStore } from "@reduxjs/toolkit";
import { createLogger } from "redux-logger";
import missionsReducer from "./missions/slice";

export default configureStore({
  reducer: {
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(createLogger()),
});
