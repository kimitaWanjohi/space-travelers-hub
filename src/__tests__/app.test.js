import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/store";
import App from "../App";

describe("App", () => {
  test("renders App component", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
});
