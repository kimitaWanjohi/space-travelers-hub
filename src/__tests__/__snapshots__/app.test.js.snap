// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <nav
    className="py-5 px-2 shadow-md"
  >
    <div
      className="container mx-auto flex items-baseline justify-between"
    >
      <div
        className="flex items-center"
      >
        <img
          alt="logo"
          className="w-10"
          src="/planet.png"
        />
        <h1
          className="text-2xl font-bold ml-2"
        >
          Space Traveler's Hub
        </h1>
      </div>
      <ul
        className="flex items-center"
      >
        <li
          className="mr-6"
        >
          <a
            aria-current="page"
            className="text-md bg-white underline text-blue-500 cursor-pointer hover:text-blue-700 active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li
          className="mr-6"
        >
          <a
            className="text-md bg-white underline text-blue-500 cursor-pointer hover:text-blue-700"
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          <a
            className="text-md bg-white underline text-blue-500 cursor-pointer hover:text-blue-700"
            href="/my-profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container mx-auto"
  >
    <div
      className="mt-6 mb-8"
    >
      <div>
        Loading...
      </div>
      <ul />
    </div>
  </div>,
]
`;
